### PR TITLE
chore: add disconnect peer dialog

### PR DESCRIPTION
**Pull request by im-adithya**
Adds DisconnectPeer to LND

Adds disconnect peer dialog, related to #419 

Fixes text in toast by adding `break-all`

## Screenshot

Screenshot 2024-08-09 at 3 55 37 PM

### CODE REVIEW

File: frontend/src/screens/peers/Peers.tsx [hunk 1]
```diff
@@ -18,20 +23,19 @@ import {
...
-import { useCSRF } from "../../hooks/useCSRF.ts";
 
 export default function Peers() {
   useSyncWallet();
-  const { data: peers, mutate: reloadPeers } = usePeers();
+  const { data: peers } = usePeers();
   const { data: channels } = useChannels();
   const [nodes, setNodes] = React.useState<Node[]>([]);
-  const { data: csrf } = useCSRF();
+  const [openPeerId, setOpenPeerId] = React.useState<string | null>(null);
```
Comment (rolznz) [inline frontend/src/screens/peers/Peers.tsx:38]: what is an `open peer`

Comment (rolznz) [reply frontend/src/screens/peers/Peers.tsx:38]: why do we need multiple dialogs? can't you just render the dialog component outside of the table?

Comment (im-adithya) [reply frontend/src/screens/peers/Peers.tsx]: I thought the Trigger is necessary for the Dialog to function, fixed now, should be good. I think this way we can also simplify the transactions list. Wil also have a look there

File: frontend/src/components/DisconnectPeerDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+type Props = {
+  peer: Peer;
+};
+
+export function DisconnectPeerDialog({ peer }: Props) {
+  const { mutate: reloadPeers } = usePeers();
+
+  async function disconnectPeer() {
+    try {
+      if (!peer.nodeId) {
```
Comment (rolznz) [inline frontend/src/components/DisconnectPeerDialog.tsx:24]: how can a peer exist but not the nodeId?

File: frontend/src/screens/peers/Peers.tsx [hunk 3]
```diff
@@ -56,31 +59,20 @@ export default function Peers() {
     loadNodeStats();
   }, [loadNodeStats]);
 
-  async function disconnectPeer(peerId: string) {
+  async function checkDisconnectPeer(peer: Peer) {
     try {
-      if (!peerId) {
+      if (!peer.nodeId) {
```
Comment (rolznz) [inline frontend/src/screens/peers/Peers.tsx:64]: this check is not needed, check the diff

File: frontend/src/screens/peers/Peers.tsx [hunk 4]
```diff
@@ -18,18 +20,19 @@ import {
...
+import { Node, Peer } from "src/types";
 import { request } from "src/utils/request";
 
 export default function Peers() {
   useSyncWallet();
-  const { data: peers, mutate: reloadPeers } = usePeers();
+  const { data: peers } = usePeers();
   const { data: channels } = useChannels();
   const [nodes, setNodes] = React.useState<Node[]>([]);
+  const [openPeer, setOpenPeer] = React.useState<Peer | null>(null);
```
Comment (rolznz) [inline frontend/src/screens/peers/Peers.tsx:35]:
```suggestion
  const [peer, setPeer] = React.useState<Peer | null>(null);
```

Comment (rolznz) [inline frontend/src/screens/peers/Peers.tsx:35]:
```suggestion
  const [openPeer, setOpenPeer] = React.useState<Peer>();
```

File: frontend/src/components/DisconnectPeerDialog.tsx (hunk 2 above)
Comment (rolznz) [reply frontend/src/components/DisconnectPeerDialog.tsx:24]: @im-adithya can you please test this? otherwise we are adding extra unnecessary code and setting up poor conventions which will be copied through the app

Comment (im-adithya) [reply frontend/src/components/DisconnectPeerDialog.tsx:24]: I had the same question, but this check was used previously so I assumed there might be an issue with one of the backends where nodeId is not returned. Should I remove this?

File: frontend/src/screens/peers/Peers.tsx (hunk 4 above)
Comment (im-adithya) [reply frontend/src/screens/peers/Peers.tsx:35]: If it can't be null then the dialog would be open always...

Comment (im-adithya) [reply frontend/src/screens/peers/Peers.tsx:35]: I thought this would be confusing with the peer in the mapping

File: frontend/src/components/DisconnectPeerDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        title: "Failed to disconnect peer: " + e,
+      });
+      console.error(e);
+    }
+  }
+
+  return (
+    <AlertDialogContent>
+      <AlertDialogHeader>
+        <AlertDialogTitle className="break-all">
```
Comment (rolznz) [inline frontend/src/components/DisconnectPeerDialog.tsx:46]: Why did you add this to 3 different places in this file? surely one is enough?

File: frontend/src/components/ui/toast.tsx [hunk 6]
```diff
@@ -92,7 +92,7 @@ const ToastTitle = React.forwardRef<
 >(({ className, ...props }, ref) => (
   <ToastPrimitives.Title
     ref={ref}
-    className={cn("text-sm font-semibold [&+div]:text-xs", className)}
+    className={cn("break-all text-sm font-semibold [&+div]:text-xs", className)}
```
Comment (rolznz) [inline frontend/src/components/ui/toast.tsx:95]: we should be careful with using `break-all`. Can you try `break-words`?

File: frontend/src/screens/peers/Peers.tsx (hunk 4 above)
Comment (rolznz) [reply frontend/src/screens/peers/Peers.tsx:35]: Unless you pass a value to `useState` then the type will be `Peer | undefined` which means the default will be `undefined`. We do this much more often than using null

File: frontend/src/components/ui/toast.tsx (hunk 6 above)
Comment (im-adithya) [reply frontend/src/components/ui/toast.tsx:95]: Yup was also concerned about "more" being broken into "m-ore". But `break-words` doesn't do the job, the problem is caused by one word (the pubkey)

**Review by rolznz — APPROVED** [review]
tACK
